packages/nvidia_nat_core/src/nat/profiler/parameter_optimization/optimizer_runtime.py

```
@@ -27,6 +27,95 @@
+        if not tracing:
+            return None
+
+        # Extract dataset name from eval config (runtime concern, not plugin-specific)
+        opt_dataset_name = None
+        try:
+            ds_cfg = base_cfg.eval.general.dataset
+            file_path = getattr(ds_cfg, 'file_path', None)
+            if file_path:
+                opt_dataset_name = Path(file_path).stem
+        except Exception:
+            pass
```

Ruff (0.15.0)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)